### PR TITLE
Fix tests and increase coverage for pjax

### PR DESCRIPTION
**Pull request by vkhramtsov**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | Partially
| Fixed issues  | Pjax coverage

### CODE REVIEW

File: tests/framework/widgets/PjaxTest.php [hunk 1]
```diff
@@ -7,35 +7,57 @@
...
+    public function testRunWithPjax()
+    {
+        $this->mockWebApplication();
+        $request = Yii::$app->getRequest();
+        $request->addHeader('X-Pjax', 1);
+        $request->addHeader('X-Pjax-Container', '#p1');
+        $request->url = '/test';
+        $result = '<![CDATA[YII-BLOCK-HEAD]]><![CDATA[YII-BLOCK-BODY-BEGIN]]>';
+        $this->expectOutputString($result);
+        $this->expectException(ExitException::class);// But why exception? I want to check output here!
```
Comment (vkhramtsov) [inline tests/framework/widgets/PjaxTest.php]: Should I handle only exception here?

File: tests/framework/widgets/PjaxTest.php [hunk 2]
```diff
@@ -7,35 +7,57 @@
...
+        $request->addHeader('X-Pjax', 1);
+        $request->addHeader('X-Pjax-Container', '#p0');
+        $request->url = '/test';
+        new Pjax();
+        $result = '<![CDATA[YII-BLOCK-HEAD]]><![CDATA[YII-BLOCK-BODY-BEGIN]]>';
+        /*
+        * Unfortunately I have to do this twice to avoid php unit message like
+        * 'Test code or tested code did not (only) close its own output buffers'
+        */
+        ob_end_flush();
```
Comment (vkhramtsov) [inline tests/framework/widgets/PjaxTest.php]: Do you know how to avoid `ob_end_flush` and repetition here?

**Comment by rob006** [comment]
PJAX is going to be removed in 2.1: https://github.com/yiisoft/yii2/wiki/Roadmap#cleanup

**Comment by vkhramtsov** [comment]
@rob006 thank you, so I can close this PR and move to next class in widgets directory 😄 ?

**Comment by samdark** [comment]
Thanks for contribution but I'm afraid we're going to drop PJAX in 2.1 altogether.

**Comment by vkhramtsov** [comment]
No problem. You can close this PR. I will proceed with tests for next class.

**Pull request closed without merging**